### DISCUSSION TITLE
Is there any way to use flatten-js without npm?

### DISCUSSION THREAD

**Discussion by momolinus** [alexbol99/flatten-js#110] [Q&A] [answered]
Hi,
I want to use flatten without npm. So in the browser: <script src="flatten-js.js"></script> Is that possible somehow?
Thanks.

Marcus

**Comment by alexbol99** [comment, marked as answer]
Hello, Marcus,

I am so sorry that I missed your question.
I hope you've already found the answer in the [examples](https://github.com/alexbol99/flatten-js/tree/master/examples) section of the repository.

You can take it as umd package from ```unpackage.com``` 
```html
<script src='https://unpkg.com/@flatten-js/core'></script>
```
or as es6 package with ```script type="module"``` tag
```html
<script type="module">
      import {point, circle, segment} from "https://unpkg.com/@flatten-js/core?module";

      // make some construction
</script>
```

**Comment by Kragrathea** [comment]
I have a similar question. I am strictly running in a browser like above, but will also need to modify the flatten-js code. Can this be done without having to publish a new package every time I make a change to the library? 

Ideally I'd like to serve flatten-js directly from the source tree during development. I am not really familiar with how Node and npm really differ from straight javascript in a browser. I tried including the "index.js" script in my html page. But but there were issues like the imports all needing to specify the .JS file extension (ie './utils/constants.JS'). I guess under node the .JS is assumed. Once I got past that there were other package dependencies like flatten/tree-interval that I am not yet sure how to fix up.  Am I going about this the right way?

Thanks!
K

**Reply by edemaine** [reply]
If you're going to modify the code, it would probably be best to build it yourself. (No need to publish.)  If you have a modified version of the repo, you can run `npm install` and then `npm build` and you should get `dist/main.*` files that you can directly include in your HTML (following similar rules to above).
